[API Compat] dotnet restore does not respect IsPackable false for package validation

### Describe the bug
When I use `Directory.Build.props` to enable package validation for all packages/projects in that directory, it should be enough to have `<IsPackable >false</IsPackable>` in a csproj to fully exempt that project from package validation. Currently this is not the case because if `<PackageValidationBaselineVersion>` is present in `Directory.Build.props`, `dotnet restore` will try to restore the package with that version even if there will be no package validation for this project later.

Generally I think it is a bit problematic that packages only need for package validation which will only run at a later time during `dotnet pack` (if at all) will already be restored during the more general `dotnet restore`.

### To Reproduce
1. Have a solution with some projects and a `Directory.Build.Props` with the following content:
   ````xml
   <Project>    
        <PropertyGroup>
            <PackageVersion>1.0.0</PackageVersion>
            <EnablePackageValidation>true</EnablePackageValidation>
            <PackageValidationBaselineVersion>1.0.0</PackageValidationBaselineVersion>
        </PropertyGroup>    
   </Project>
   ```
2. Pack and push the packages to the common feed.
3. Add a new project where `IsPackgable` is `false`:
   ```xml
    <Project Sdk="Microsoft.NET.Sdk">    
        <PropertyGroup>
            <TargetFramework>net6.0</TargetFramework>
            <ImplicitUsings>enable</ImplicitUsings>
            <Nullable>enable</Nullable>
            <IsPackable>false</IsPackable>
        </PropertyGroup>    
    </Project>
   ```
4. Run `dotnet restore`

#### Actual Result
The following error occurs
```
 C:\source\AdventureWorks.Client\AdventureWorks.Utils\AdventureWorks.Utils.csproj : error NU1101: Unable to find package AdventureWorks.Utils. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, nuget.org, localnugetfeed
```

#### Expected Result

No error occurs because `IsPackgable` is `false` and therefore a baseline package will never be needed and should not be restored.

#### Known workarounds

- Explicitly disable package validation using `<EnablePackageValidation>false</EnablePackageValidation>` in the csproj.
- Reset the baseline version `<PackageValidationBaselineVersion />` in the csproj.

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ApiCompat.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ApiCompat.targets
@@ -16,6 +16,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <ItemGroup Condition="'$(EnablePackageValidation)' == 'true' and
+                        '$(IsPackable)' == 'true' and
                         '$(DisablePackageBaselineValidation)' != 'true' and
                         '$(PackageValidationBaselinePath)' == '' and
                         '$(PackageValidationBaselineVersion)' != ''">

